pkg/services/alerting/eval_context.go

```diff
@@ -146,6 +146,20 @@ func (c *EvalContext) GetNewState() models.AlertStateType {
 	return models.AlertStatePending
 }
 
+func (c *EvalContext) ToggleNoDataFlag() bool {
```

Comment:
I find the naming of this a bit weird since you actually don't toggle anything. It's similar to `GetNewState`, but naming it `GetNewNoDataFlag` is also weird.

Reply:
Another option would be `UpdateNoDataFlag` but I think that  `GetNewNoDataFlag` is better since the function only returns the new value but does not alter anything.

Reply:
anyway even `NoDataFlag` is not the best choice but I couldn't think of anything better.